### ISSUE TITLE
Scheduled repository backup / restore feature

### ISSUE THREAD

**Issue by milkyway0308** [dzikoysk/reposilite#1082]
### Request details

Support repository backup system repeating every n hours, and backup file import feature.

**Comment by dzikoysk** [comment]
Yeah, we don't have any kind of backup system, good idea to consider at least a basic one.

I think we're mostly considering only local infrastructure such as file system & sqlite? I guess it's up to the user to maintain backups of external components like AWS S3/other remote DBs? 

Also, thanks for your support! :)

**Comment by codesplode** [comment]
Not sure what Javalin has in the way of scheduling, but two paths that come to mind:
- A very simplistic backup using coroutines on a delayed check
- A much more flexible and configurable solution using [Quartz](http://www.quartz-scheduler.org/documentation/quartz-2.3.0/quick-start.html) which would allow user scheduling and UI based customization later as the schedules are very configurable, handle "misfires" (when a task is missed, etc.). The library itself is pretty lightweight, but the first use can feel a little heavy since you have so many options with it. Just using the file system and no other fancy stores should be a pretty quick setup.

**Comment by dzikoysk** [comment]
Scheduling itself is not a problem, it's more likely the way how we'd like to archive things, especially having multiple infrastructure targets. In general I think that Reposilite has pretty straightforward file structure that you can just backup as a whole using external tools like even such simple thing like a cron.

**Comment by codesplode** [comment]
Sure. As long as it's safe mid transaction I think that would be perfectly fine. I'm not super familiar with all the things reposilite uses the database for yet. Other tools would primarily help across platform where tools like cron are not always available. Regardless, I'm always on board with keeping it light.
